### DISCUSSION TITLE
ScriptReference/AssetPostprocessor.OnPreprocessModel

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#598] [Notes]
# ScriptReference/AssetPostprocessor.OnPreprocessModel

https://docs.unity3d.com/ScriptReference/AssetPostprocessor.OnPreprocessModel.html

**Comment by lazlo-bonin** [comment]
Note that this is called even on models that already have a meta file, so it might overwrite your existing settings:
https://forum.unity.com/threads/under-what-conditions-is-onpreprocessmodel-called.985486/

The detection for whether it should run seems to only check local library artifacts, not meta file existence, making it unreliable to apply default settings.
